src/Common/src/System/Security/Cryptography/Asn1V2.cs

```diff
@@ -0,0 +1,5071 @@
+
+                    if (write)
+                    {
+                        destination = destination.Slice(pretendWritten);
+                    }
+                }
+            }
+
+            if (isIndefinite)
+            {
+                throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+            }
+
+            bytesRead = totalRead;
+            return totalContent;
+        }
+
+        private static bool TryCopyConstructedBitStringValue(
+            ReadOnlyMemory<byte> source,
+            Span<byte> dest,
+            AsnEncodingRules ruleSet,
+            bool isIndefinite,
+            out int unusedBitCount,
+            out int bytesRead,
+            out int bytesWritten)
+        {
+            int lastUnusedBitCount = 0;
+            int lastSegmentSize = MaxCERSegmentSize;
+
+            Span<byte> tmpDest = dest;
+
+            int contentLength = CopyConstructedBitString(
+                source,
+                ref tmpDest,
+                false,
+                ruleSet,
+                isIndefinite,
+                ref lastUnusedBitCount,
+                ref lastSegmentSize,
+                out int encodedLength);
+
+            // Since the unused bits byte from the segments don't count, only one segment
+            // returns 999 (or less), the second segment bumps the count to 1000, and is legal.
+            //
+            // T-REC-X.690-201508 sec 9.2
+            if (ruleSet == AsnEncodingRules.CER && contentLength < MaxCERSegmentSize)
+            {
+                throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+            }
+
+            if (dest.Length < contentLength)
+            {
+                unusedBitCount = 0;
+                bytesRead = 0;
+                bytesWritten = 0;
+                return false;
+            }
+
+            tmpDest = dest;
+            unusedBitCount = lastUnusedBitCount;
+            lastSegmentSize = MaxCERSegmentSize;
+            lastUnusedBitCount = 0;
+
+            bytesWritten = CopyConstructedBitString(
+                source,
+                ref tmpDest,
+                true,
+                ruleSet,
+                isIndefinite,
+                ref lastUnusedBitCount,
+                ref lastSegmentSize,
+                out bytesRead);
+
+            Debug.Assert(unusedBitCount == lastUnusedBitCount);
+            Debug.Assert(bytesWritten == contentLength);
+            Debug.Assert(bytesRead == encodedLength);
+            return true;
+        }
+
+        private bool TryGetBitStringBytes(
+            Asn1Tag expectedTag,
+            out int unusedBitCount,
+            out ReadOnlyMemory<byte> contents,
+            out int headerLength)
+        {
+            (Asn1Tag tag, int? length) = ReadTagAndLength(out headerLength);
+            CheckExpectedTag(tag, expectedTag, UniversalTagNumber.BitString);
+
+            if (tag.IsConstructed)
```

Comment:
Consider a name/comment that clarifies that this can only return contiguous bytes since both versions ultimately return bytes.